src/Compilers/CSharp/Portable/FlowAnalysis/AbstractFlowPass.cs

```diff
@@ -1513,23 +1513,29 @@ public override BoundNode VisitDelegateCreationExpression(BoundDelegateCreationE
             var methodGroup = node.Argument as BoundMethodGroup;
             if (methodGroup != null)
             {
-                if ((object)node.MethodOpt != null && node.MethodOpt.RequiresInstanceReceiver)
+                if (node.MethodOpt?.OriginalDefinition is LocalFunctionSymbol localFunc)
+                {
+                    VisitLocalFunctionUse(localFunc, node.Syntax, isCall: false);
+                }
+                else if (node.MethodOpt is { } method && methodGroup.ReceiverOpt is { } receiver && !ignoreReceiver(receiver, method))
                 {
                     EnterRegionIfNeeded(methodGroup);
                     VisitRvalue(methodGroup.ReceiverOpt);
                     LeaveRegionIfNeeded(methodGroup);
                 }
-                else if (node.MethodOpt?.OriginalDefinition is LocalFunctionSymbol localFunc)
-                {
-                    VisitLocalFunctionUse(localFunc, node.Syntax, isCall: false);
-                }
             }
             else
             {
                 VisitRvalue(node.Argument);
             }
 
             return null;
+
+            static bool ignoreReceiver(BoundExpression receiver, MethodSymbol method)
+            {
+                // ignore the implicit `this` receiver on a static method
+                return method.IsStatic && receiver is { Kind: BoundKind.ThisReference, WasCompilerGenerated: true };
```

Comment:
Perhaps we should add a comment here:
we may need to visit the receiver / can't ignore static methods at all because of extension methods.